File: cltch-ntwrk/native-ios-swift/CLTCHNTWRKSwift/Services/DatabaseContract.swift
import Foundation

enum FirestoreCollection: String {
    case users
    case userRoles
    case hosts
    case musicians
    case gigs

    var path: String { rawValue }
}

struct FirebaseProjectConfig {
    let projectId = "cltch-ntwrk"
    let authDomain = "cltch-ntwrk.firebaseapp.com"
    let storageBucket = "cltch-ntwrk.firebasestorage.app"
}

enum NativeSyncNotes {
    static let roleResolutionOrder = [
        "users/{uid}",
        "userRoles/{uid}"
    ]

    static let bookingSource = "gigs/{gigId}"
    static let hostProfileSource = "hosts/{uid}"
    static let musicianProfileSource = "musicians/{uid}"
}

File: cltch-ntwrk/native-android-kotlin/app/src/main/java/com/cltch/ntwrk/nativeapp/data/DatabaseContract.kt
package com.cltch.ntwrk.nativeapp.data

object DatabaseContract {
    const val PROJECT_ID = "cltch-ntwrk"
    const val AUTH_DOMAIN = "cltch-ntwrk.firebaseapp.com"
    const val STORAGE_BUCKET = "cltch-ntwrk.firebasestorage.app"

    object Collections {
        const val USERS = "users"
        const val USER_ROLES = "userRoles"
        const val HOSTS = "hosts"
        const val MUSICIANS = "musicians"
        const val GIGS = "gigs"
    }

    val ROLE_RESOLUTION_ORDER = listOf(
        "users/{uid}",
        "userRoles/{uid}"
    )
}
